### PR TITLE
Iap manager script null product type fixed/

### PR DESCRIPTION
**Pull request by gmYusuf**
IAP Manager script updated.
User can use any type of product.
User dont need to fill all types

### CODE REVIEW

File: Assets/Huawei/Scripts/IAP/IapManager.cs [hunk 1]
```diff
@@ -132,7 +132,87 @@ public void ObtainProductInfo(IList<string> productIdConsumablesList, IList<stri
...
+                Debug.Log("[HMSPlugin]:" + type0.ErrMsg + type0.ReturnCode.ToString());
+                Debug.Log("[HMSPlugin]: Found " + type0.ProductInfoList.Count + "consumable products");
+                OnObtainProductInfoSuccess?.Invoke(new List<ProductInfoResult> { type0 });
+            }).AddOnFailureListener((exception) =>
+            {
+                Debug.Log("[HMSPlugin]: ERROR Consumable ObtainInfo" + exception.Message);
+                OnObtainProductInfoFailure?.Invoke(exception);
+
+            });
+        }
```
Comment (mmontesinos81) [inline Assets/Huawei/Scripts/IAP/IapManager.cs:161]: As code is almost the same in all methods, can you make a single method with type parameter?

File: Assets/Huawei/Demos/IapDemo.unity [hunk 2]
```diff
@@ -139,6 +139,49 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: b10819352ed36d84c88e5d57c5dd9b9a, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
+--- !u!1 &282388502
```
[the remark this answers is not in the document]
Comment (gmYusuf) [reply Assets/Huawei/Demos/IapDemo.unity:142]: I didn't change something. We don't need to update this.

File: Assets/Huawei/Demos/IAP/IapDemoManager.cs [hunk 3]
```diff
@@ -91,8 +91,18 @@ private void LoadStore()
...
+        }
+        if (!IsNullOrEmpty(NonConsumableProducts))
+        {
+            iapManager.ObtainProductNonConsumablesInfo(new List<string>(NonConsumableProducts));
+        }
+        if (!IsNullOrEmpty(SubscriptionProducts))
+        {
+            iapManager.ObtainProductSubscriptionInfo(new List<string>(SubscriptionProducts));
+        }
     }
```
[the remark this answers is not in the document]
Comment (gmYusuf) [reply Assets/Huawei/Demos/IAP/IapDemoManager.cs:106]: I will update this.

File: Assets/Huawei/Scripts/IAP/IapManager.cs (hunk 1 above)
Comment (gmYusuf) [reply Assets/Huawei/Scripts/IAP/IapManager.cs:161]: I will fix this.

File: Assets/Huawei/Demos/IapDemo.unity (hunk 2 above)
[the remark this answers is not in the document]
Comment (m0skit0) [reply Assets/Huawei/Demos/IapDemo.unity:142]: Then please revert this change in your PR :)

**Pull request closed without merging**